ddtrace/_trace/pin.py

```diff
@@ -0,0 +1,237 @@
+        instance, avoiding that a specific instance overrides other pins values.
+
+            >>> pin = Pin.get_from(conn)
+
+        :param obj: The object to look for a :class:`ddtrace.pin.Pin` on
+        :type obj: object
+        :rtype: :class:`ddtrace.pin.Pin`, None
+        :returns: :class:`ddtrace.pin.Pin` associated with the object, or None if none was found
+        """
+        if hasattr(obj, "__getddpin__"):
+            return obj.__getddpin__()
+
+        pin_name = _DD_PIN_PROXY_NAME if isinstance(obj, wrapt.ObjectProxy) else _DD_PIN_NAME
+        pin = getattr(obj, pin_name, None)
+        # detect if the PIN has been inherited from a class
+        if pin is not None and pin._target != id(obj):
+            pin = pin.clone()
+            pin.onto(obj)
+        return pin
+
+    @classmethod
+    def override(
+        cls,
+        obj,  # type: Any
+        service=None,  # type: Optional[str]
+        tags=None,  # type: Optional[Dict[str, str]]
+        tracer=None,
+    ):
+        # type: (...) -> None
+        """Override an object with the given attributes.
+
+        That's the recommended way to customize an already instrumented client, without
+        losing existing attributes.
+
+            >>> conn = sqlite.connect('/tmp/user.db')
+            >>> # Override a pin for a specific connection
+            >>> Pin.override(conn, service='user-db')
+        """
+        if not obj:
+            return
+
+        pin = cls.get_from(obj)
+        if pin is None:
+            Pin(service=service, tags=tags, tracer=tracer)._onto(obj)
+        else:
+            pin._clone(service=service, tags=tags, tracer=tracer)._onto(obj)
+
+    def enabled(self):
+        # type: () -> bool
+        """Return true if this pin's tracer is enabled."""
+        return bool(self.tracer) and (self.tracer.enabled or self.tracer._apm_opt_out)
+
+    def _onto(self, obj, send=True):
+        # type: (Any, bool) -> None
+        """Patch this pin onto the given object. If send is true, it will also
+        queue the metadata to be sent to the server.
+        """
+        # Actually patch it on the object.
+        try:
+            if hasattr(obj, "__setddpin__"):
+                return obj.__setddpin__(self)
+
+            pin_name = _DD_PIN_PROXY_NAME if isinstance(obj, wrapt.ObjectProxy) else _DD_PIN_NAME
+
+            # set the target reference; any get_from, clones and retarget the new PIN
+            self._target = id(obj)
+            if self.service:
+                ddtrace.config._add_extra_service(self.service)
+            return setattr(obj, pin_name, self)
+        except AttributeError:
+            log.debug("can't pin onto object. skipping", exc_info=True)
+
+    def onto(self, obj, send=True):
+        deprecate(
+            "The `pin.onto` method is deprecated and will be removed.",
+            message="""The `pin.onto` method is deprecated and will be removed.""",
```

Comment:
We should also inform users why this method is being deprecated and if there is an alternative. Users could be getting value from this functionality.